### PR TITLE
Regression

### PR DESCRIPTION
**Pull request by m-ghosal**
Implemented leveling algorithm and implemented tests for the leveling algorithm.

### CODE REVIEW

File: tests/skilltrees/skill_test.c [hunk 1]
```diff
@@ -76,35 +76,33 @@ Test(skill_tests, skill_execute_test)
...
 /* Tests skill_level_up when return value is 0 */
-
-Test(skill_tests, skill_level_up_0) {
-    skill_t* skill = skill_new(1000, ACTIVE, "defuse bomb", "defuses a bomb",
-      2, 5, effect_defuse_bomb);
-    check_level_up(skill, 0);
+Test(skill_tests, skill_level_up_0)
+{
+  skill_t* skill = skill_new(1000, ACTIVE, "defuse bomb", "defuses a bomb",
+    2, 5, effect_defuse_bomb);
```
Comment (Joshuashou) [inline tests/skilltrees/skill_test.c:90]: are these the same change?

**Review by rohangupta2806 — APPROVED** [review]
I went through the changes and they seem to make sense.  I don't think anything needs to be changed.  Approved.

File: tests/skilltrees/skill_test.c (hunk 1 above)
Comment (m-ghosal) [reply tests/skilltrees/skill_test.c:98]: no this was what i added

**Review by firatciftci — APPROVED** [review]
Everything looks good to me!

**Comment by firatciftci** [comment]
Issue Score: ✔️-

Comments:
Good work overall! I have a few things to comment on:
- The title of a PR has to convey what the issue will be producing. In the future, try using simple titles with a "Verb-Object" structure that clearly states what you was done in the PR (e.g., “Design and implementation of ‘circle’ module”, etc.).
- Even though you can add labels to PRs to categorize them, you should avoid titles that could plausibly apply to multiple teams.
- The summary for this PR is too terse. A good rule of thumb is that you should be writing at least one full paragraph in your summary (see https://uchicago-cs.github.io/cmsc22000/project/policies.html#tasks-issues for examples).
- When an PR is done, you should include a final comment summarizing what was merged to `dev` and what work remains to be done, if any.

**Pull request merged**